### ISSUE TITLE
client panics when heketi config has a syntax error

### ISSUE THREAD

**Issue by obnoxxx** [heketi/heketi#464]
Seen in the vagrant-heketi standalone setup:
prereq: heketi.json with a syntax error, like forgot a comma in a list.

```
$ systemctl start heketi
$ export HEKETI_CLI_SERVER=http://localhost:8080
$ heketi-cli topology load -j ./topology_libvirt.json 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x469d25]

goroutine 1 [running]:
panic(0xc6b3c0, 0xc8200100c0)
        /usr/lib/golang/src/runtime/panic.go:481 +0x3e6
github.com/heketi/heketi/client/cli/go/cmds.getNodeIdFromHeketiTopology(0x0, 0xc8201cc690, 0xe, 0xc8201ad8f0)
        /builddir/build/BUILD/heketi-66f7b09/src/github.com/heketi/heketi/client/cli/go/cmds/topology.go:70 +0x2c5
github.com/heketi/heketi/client/cli/go/cmds.glob.func17(0x13f5800, 0xc820118b40, 0x0, 0x2, 0x0, 0x0)
        /builddir/build/BUILD/heketi-66f7b09/src/github.com/heketi/heketi/client/cli/go/cmds/topology.go:137 +0x89b
github.com/spf13/cobra.(*Command).execute(0x13f5800, 0xc8201189e0, 0x2, 0x2, 0x0, 0x0)
        /builddir/build/BUILD/heketi-66f7b09/Godeps/_workspace/src/github.com/spf13/cobra/command.go:568 +0x62c
github.com/spf13/cobra.(*Command).ExecuteC(0x13f5400, 0x13f5800, 0x0, 0x0)
        /builddir/build/BUILD/heketi-66f7b09/Godeps/_workspace/src/github.com/spf13/cobra/command.go:662 +0x53f
github.com/spf13/cobra.(*Command).Execute(0x13f5400, 0x0, 0x0)
        /builddir/build/BUILD/heketi-66f7b09/Godeps/_workspace/src/github.com/spf13/cobra/command.go:618 +0x2d
main.main()
        /builddir/build/BUILD/heketi-66f7b09/client/cli/go/main.go:35 +0xe0
```

**Comment by obnoxxx** [comment]
Oh, this is a duplicate of #445 !

**Comment by obnoxxx** [comment]
The strange thing though is that while the panic looks like the topology-file panic, the syntax error was actually in the hekti config file... :-o

**Comment by lpabon** [comment]
@obnoxxx Do you still see this error?  I cannot reproduce:

```
$ ./heketi-cli -s http://localhost:8080 topology load --json=bad.json
Error: Unable to parse config file
```

**Comment by lpabon** [comment]
@obnoxxx ping

**Comment by obnoxxx** [comment]
Ok, the problem (as I described it in the initial report) exists exactly the same way still.

The point was not the syntax error in the json file leading to panic though:
Heketi-cli behaves perfectly ok when the config file has a syntax error.
The problem was that the heketi server refused to start upon erroneous config file (heketi.json), and
the client was panic-ing over the non-reachable server.

**Comment by obnoxxx** [comment]
that is .... the problem exists in the heketi-cli used in the vagrant-heketi test setup

**Comment by obnoxxx** [comment]
So this was actually fixed by the patches for issue #445 the point was that this patch fixes ANY previously missed error propagation from TopologyInfo(), including one which was caused by error to connect to the server.
